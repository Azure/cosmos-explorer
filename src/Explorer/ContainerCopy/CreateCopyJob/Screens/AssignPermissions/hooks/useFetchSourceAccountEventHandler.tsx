import { useCallback, useState } from "react";
import { fetchDatabaseAccount } from "../../../../../../Utils/arm/databaseAccountUtils";
import { useCopyJobContext } from "../../../../Context/CopyJobContext";
import { getAccountDetailsFromResourceId } from "../../../../CopyJobUtils";

const useFetchSourceAccountEventHandler = () => {
  const [loading, setLoading] = useState<boolean>(false);
  const { copyJobState: { source } = {}, setCopyJobState } = useCopyJobContext();
  const onWindowClosed = useCallback(async (errorMessage: string) => {
    try {
      const selectedSourceAccount = source?.account;
      const {
        subscriptionId: sourceSubscriptionId,
        resourceGroup: sourceResourceGroup,
        accountName: sourceAccountName,
      } = getAccountDetailsFromResourceId(selectedSourceAccount?.id);

      setLoading(true);
      const account = await fetchDatabaseAccount(sourceSubscriptionId, sourceResourceGroup, sourceAccountName);
      if (account) {
        setCopyJobState((prevState) => ({
          ...prevState,
          source: { ...prevState.source, account: account },
        }));
      }
    } catch (error) {
      console.error(errorMessage, error);
    } finally {
      setLoading(false);
    }
  }, []);

  return { loading, onWindowClosed };
};

export default useFetchSourceAccountEventHandler;
