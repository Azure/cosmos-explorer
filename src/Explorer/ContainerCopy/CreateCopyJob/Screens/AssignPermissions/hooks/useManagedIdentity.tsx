import { DatabaseAccount } from "Contracts/DataModels";
import { useCallback, useState } from "react";
import { useCopyJobContext } from "../../../../Context/CopyJobContext";
import { getAccountDetailsFromResourceId } from "../../../../CopyJobUtils";

interface UseManagedIdentityUpdaterParams {
    updateIdentityFn: (
        subscriptionId: string,
        resourceGroup?: string,
        accountName?: string
    ) => Promise<DatabaseAccount | undefined>;
}

interface UseManagedIdentityUpdaterReturn {
    loading: boolean;
    handleAddSystemIdentity: () => Promise<void>;
}

const useManagedIdentity = (
    updateIdentityFn: UseManagedIdentityUpdaterParams["updateIdentityFn"]
): UseManagedIdentityUpdaterReturn => {
    const { copyJobState, setCopyJobState } = useCopyJobContext();
    const [loading, setLoading] = useState<boolean>(false);

    const handleAddSystemIdentity = useCallback(async (): Promise<void> => {
        try {
            setLoading(true);
            const selectedTargetAccount = copyJobState?.target?.account;
            const {
                subscriptionId: targetSubscriptionId,
                resourceGroup: targetResourceGroup,
                accountName: targetAccountName
            } = getAccountDetailsFromResourceId(selectedTargetAccount?.id);

            const updatedAccount = await updateIdentityFn(
                targetSubscriptionId,
                targetResourceGroup,
                targetAccountName
            );
            if (updatedAccount) {
                setCopyJobState((prevState) => ({
                    ...prevState,
                    target: { ...prevState.target, account: updatedAccount }
                }));
            }
        } catch (error) {
            console.error("Error enabling system-assigned managed identity:", error);
        } finally {
            setLoading(false);
        }
    }, [copyJobState, updateIdentityFn, setCopyJobState]);

    return { loading, handleAddSystemIdentity };
};

export default useManagedIdentity;