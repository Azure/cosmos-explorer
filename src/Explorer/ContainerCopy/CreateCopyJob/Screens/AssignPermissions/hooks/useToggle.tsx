import { useCallback, useState } from "react";

const useToggle = (initialState = false) => {
  const [state, setState] = useState<boolean>(initialState);
  const onToggle = useCallback((_, checked?: boolean) => {
    setState(checked);
  }, []);
  return [state, onToggle] as const;
};

export default useToggle;
