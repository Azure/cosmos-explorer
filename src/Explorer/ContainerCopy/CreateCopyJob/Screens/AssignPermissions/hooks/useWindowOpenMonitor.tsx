import { useEffect, useRef } from "react";

const useWindowOpenMonitor = (url: string, onClose?: () => void, intervalMs = 500) => {
    const intervalRef = useRef<NodeJS.Timeout | null>(null);

    const openWindowAndMonitor = () => {
        const newWindow = window.open(url, '_blank');
        intervalRef.current = setInterval(() => {
            if (newWindow?.closed) {
                clearInterval(intervalRef.current!);
                intervalRef.current = null;
                console.log('New window has been closed!');
                if (onClose) {
                    onClose();
                }
            }
        }, intervalMs);
    };

    useEffect(() => {
        return () => {
            if (intervalRef.current) {
                clearInterval(intervalRef.current);
                intervalRef.current = null;
            }
        };
    }, []);

    return openWindowAndMonitor;
};

export default useWindowOpenMonitor;