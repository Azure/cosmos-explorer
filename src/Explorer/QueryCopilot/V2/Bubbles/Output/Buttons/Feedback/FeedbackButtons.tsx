import { Callout, DirectionalHint, IconButton, Link, Stack, Text } from "@fluentui/react";
import { useId } from "@fluentui/react-hooks";
import { useQueryCopilot } from "hooks/useQueryCopilot";
import React, { useState } from "react";
import LikeHover from "../../../../../../../../images/CopilotLikeHover.svg";
import LikePressed from "../../../../../../../../images/CopilotLikePressed.svg";
import LikeRest from "../../../../../../../../images/CopilotLikeRest.svg";

export const FeedbackButtons: React.FC = (): JSX.Element => {
  const { generatedQuery, userPrompt } = useQueryCopilot();

  const [likeQuery, setLikeQuery] = useState<boolean>(false);
  const [dislikeQuery, setDislikeQuery] = useState<boolean>(false);
  const [likeImageLink, setLikeImageLink] = useState<string>(LikeRest);
  const [dislikeImageLink, setDislikeImageLink] = useState<string>(LikeRest);
  const [calloutVisible, setCalloutVisible] = useState<boolean>(false);
  const likeBtnId = useId("likeBtn");
  const dislikeBtnId = useId("dislikeBtn");

  return (
    <Stack horizontal>
      {calloutVisible && (
        <Callout
          target={`#${likeBtnId}`}
          onDismiss={() => setCalloutVisible(false)}
          directionalHint={DirectionalHint.topCenter}
          role="dialog"
          style={{ padding: "5px 12px 5px 12px", borderRadius: "4px" }}
          styles={{ beakCurtain: { borderRadius: "4px" }, root: { borderRadius: "4px" } }}
        >
          <Text>
            {" "}
            <Text>
              Thank you. Need to give{" "}
              <Link
                onClick={() => {
                  setCalloutVisible(false);
                  useQueryCopilot.getState().openFeedbackModal(generatedQuery, true, userPrompt);
                }}
              >
                more feedback?
              </Link>
            </Text>
          </Text>
        </Callout>
      )}

      <IconButton
        id={likeBtnId}
        iconProps={{
          imageProps: { src: likeImageLink },
          style: { minHeight: "18px" },
        }}
        onClick={() => {
          if (likeQuery) {
            setLikeQuery(false);
            setLikeImageLink(LikeRest);
            setCalloutVisible(false);
          } else {
            setLikeQuery(true);
            setDislikeQuery(false);
            setLikeImageLink(LikePressed);
            setDislikeImageLink(LikeRest);
            setCalloutVisible(true);
          }
        }}
        onMouseOver={() => setLikeImageLink(LikeHover)}
        onMouseLeave={() => setLikeImageLink(likeQuery ? LikePressed : LikeRest)}
      />
      <IconButton
        id={dislikeBtnId}
        iconProps={{
          imageProps: { src: dislikeImageLink },
          style: { minHeight: "18px", transform: "rotate(180deg)" },
        }}
        onClick={() => {
          if (dislikeQuery) {
            setDislikeQuery(false);
            setDislikeImageLink(LikeRest);
          } else {
            setDislikeQuery(true);
            setLikeQuery(false);
            setDislikeImageLink(LikePressed);
            setLikeImageLink(LikeRest);
            useQueryCopilot.getState().openFeedbackModal(generatedQuery, false, userPrompt);
          }
        }}
        onMouseOver={() => setDislikeImageLink(LikeHover)}
        onMouseLeave={() => setDislikeImageLink(dislikeQuery ? LikePressed : LikeRest)}
      />
    </Stack>
  );
};
