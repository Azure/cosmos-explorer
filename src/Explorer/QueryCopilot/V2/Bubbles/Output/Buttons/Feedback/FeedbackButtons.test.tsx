import { Callout, IconButton, Link } from "@fluentui/react";
import { useId } from "@fluentui/react-hooks";
import { FeedbackButtons } from "Explorer/QueryCopilot/V2/Bubbles/Output/Buttons/Feedback/FeedbackButtons";
import { shallow } from "enzyme";
import { useQueryCopilot } from "hooks/useQueryCopilot";
import React from "react";
import LikeHover from "../../../../../../../../images/CopilotLikeHover.svg";
import LikePressed from "../../../../../../../../images/CopilotLikePressed.svg";
import LikeRest from "../../../../../../../../images/CopilotLikeRest.svg";

useId as jest.Mock;

jest.mock("../../../../../../../../images/CopilotLikeHover.svg", () => "LikeHover");
jest.mock("../../../../../../../../images/CopilotLikePressed.svg", () => "LikePressed");
jest.mock("../../../../../../../../images/CopilotLikeRest.svg", () => "LikeRest");

beforeEach(() => {
  jest.resetAllMocks();
});

describe("Feedback buttons snapshot tests", () => {
  it("should click like and show callout", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let likeButton = wrapper.find(IconButton).first();
    const dislikeButton = wrapper.find(IconButton).last();
    likeButton.simulate("click");
    likeButton = wrapper.find(IconButton).first();
    const callout = wrapper.find(Callout).first();

    expect(likeButton.props().iconProps.imageProps.src).toEqual(LikePressed);
    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikeRest);
    expect(callout.exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should click like and dismiss callout", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    const likeButton = wrapper.find(IconButton).first();
    likeButton.simulate("click");
    let callout = wrapper.find(Callout).first();
    callout.simulate("dismiss");
    callout = wrapper.find(Callout).first();

    expect(callout.exists()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should click like and submit feedback", () => {
    const spy = jest.spyOn(useQueryCopilot.getState(), "openFeedbackModal");
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    const likeButton = wrapper.find(IconButton).first();
    likeButton.simulate("click");
    const link = wrapper.find(Link).first();
    link.simulate("click");

    expect(spy).toHaveBeenNthCalledWith(1, "", true, "");
    expect(wrapper).toMatchSnapshot();
  });

  it("should hover over like", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let likeButton = wrapper.find(IconButton).first();
    likeButton.simulate("mouseover");
    likeButton = wrapper.find(IconButton).first();

    expect(likeButton.props().iconProps.imageProps.src).toEqual(LikeHover);
    expect(wrapper).toMatchSnapshot();
  });

  it("should hover over rest like and leave", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let likeButton = wrapper.find(IconButton).first();
    likeButton.simulate("mouseover");
    likeButton.simulate("mouseleave");
    likeButton = wrapper.find(IconButton).first();

    expect(likeButton.props().iconProps.imageProps.src).toEqual(LikeRest);
    expect(wrapper).toMatchSnapshot();
  });

  it("should hover over pressed like and leave", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let likeButton = wrapper.find(IconButton).first();
    likeButton.simulate("click");
    likeButton = wrapper.find(IconButton).first();
    likeButton.simulate("mouseover");
    likeButton.simulate("mouseleave");

    expect(likeButton.props().iconProps.imageProps.src).toEqual(LikePressed);
    expect(wrapper).toMatchSnapshot();
  });

  it("should hover over like and click", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let likeButton = wrapper.find(IconButton).first();
    likeButton.simulate("mouseover");
    likeButton.simulate("click");
    likeButton = wrapper.find(IconButton).first();

    expect(likeButton.props().iconProps.imageProps.src).toEqual(LikePressed);
    expect(wrapper).toMatchSnapshot();
  });

  it("should dobule click on like", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let likeButton = wrapper.find(IconButton).first();
    likeButton.simulate("click");
    likeButton = wrapper.find(IconButton).first();
    expect(likeButton.props().iconProps.imageProps.src).toEqual(LikePressed);

    likeButton.simulate("click");
    likeButton = wrapper.find(IconButton).first();
    expect(likeButton.props().iconProps.imageProps.src).toEqual(LikeRest);
    expect(wrapper).toMatchSnapshot();
  });

  it("should click dislike and show popup", () => {
    const spy = jest.spyOn(useQueryCopilot.getState(), "openFeedbackModal");
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    const likeButton = wrapper.find(IconButton).first();
    let dislikeButton = wrapper.find(IconButton).last();
    dislikeButton.simulate("click");
    const callout = wrapper.find(Callout).first();
    dislikeButton = wrapper.find(IconButton).last();

    expect(likeButton.props().iconProps.imageProps.src).toEqual(LikeRest);
    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikePressed);
    expect(spy).toHaveBeenNthCalledWith(1, "", false, "");
    expect(callout.exists()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should hover over dislike", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let dislikeButton = wrapper.find(IconButton).last();
    dislikeButton.simulate("mouseover");
    dislikeButton = wrapper.find(IconButton).last();

    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikeHover);
    expect(wrapper).toMatchSnapshot();
  });

  it("should hover over rest dislike and leave", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let dislikeButton = wrapper.find(IconButton).last();
    dislikeButton.simulate("mouseover");
    dislikeButton.simulate("mouseleave");
    dislikeButton = wrapper.find(IconButton).last();

    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikeRest);
    expect(wrapper).toMatchSnapshot();
  });

  it("should hover over pressed dislike and leave", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let dislikeButton = wrapper.find(IconButton).last();
    dislikeButton.simulate("click");
    dislikeButton = wrapper.find(IconButton).last();
    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikePressed);
    dislikeButton.simulate("mouseover");
    dislikeButton.simulate("mouseleave");
    dislikeButton = wrapper.find(IconButton).last();

    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikePressed);
    expect(wrapper).toMatchSnapshot();
  });

  it("should hover over dislike and click", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let dislikeButton = wrapper.find(IconButton).last();
    dislikeButton.simulate("mouseover");
    dislikeButton.simulate("click");
    dislikeButton = wrapper.find(IconButton).last();

    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikePressed);
    expect(wrapper).toMatchSnapshot();
  });

  it("should dobule click on dislike", () => {
    const wrapper = shallow(<FeedbackButtons sqlQuery={""} />);

    let dislikeButton = wrapper.find(IconButton).last();
    dislikeButton.simulate("click");
    dislikeButton = wrapper.find(IconButton).last();
    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikePressed);

    dislikeButton.simulate("click");
    dislikeButton = wrapper.find(IconButton).last();
    expect(dislikeButton.props().iconProps.imageProps.src).toEqual(LikeRest);
    expect(wrapper).toMatchSnapshot();
  });
});
