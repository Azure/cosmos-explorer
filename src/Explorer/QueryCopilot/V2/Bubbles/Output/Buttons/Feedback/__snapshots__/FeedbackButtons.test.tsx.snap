// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback buttons snapshot tests should click dislike and show popup 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn16"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikePressed",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn17"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should click like and dismiss callout 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikePressed",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn2"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn3"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should click like and show callout 1`] = `
<Stack
  horizontal={true}
>
  <Callout
    directionalHint={1}
    onDismiss={[Function]}
    role="dialog"
    style={
      Object {
        "borderRadius": "4px",
        "padding": "5px 12px 5px 12px",
      }
    }
    styles={
      Object {
        "beakCurtain": Object {
          "borderRadius": "4px",
        },
        "root": Object {
          "borderRadius": "4px",
        },
      }
    }
    target="#likeBtn0"
  >
    <Text>
       
      <Text>
        Thank you. Need to give
         
        <StyledLinkBase
          onClick={[Function]}
        >
          more feedback?
        </StyledLinkBase>
      </Text>
    </Text>
  </Callout>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikePressed",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn0"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn1"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should click like and submit feedback 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikePressed",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn4"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn5"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should dobule click on dislike 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn26"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn27"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should dobule click on like 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn14"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn15"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should hover over dislike 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn18"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeHover",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn19"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should hover over dislike and click 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn24"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikePressed",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn25"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should hover over like 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeHover",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn6"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn7"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should hover over like and click 1`] = `
<Stack
  horizontal={true}
>
  <Callout
    directionalHint={1}
    onDismiss={[Function]}
    role="dialog"
    style={
      Object {
        "borderRadius": "4px",
        "padding": "5px 12px 5px 12px",
      }
    }
    styles={
      Object {
        "beakCurtain": Object {
          "borderRadius": "4px",
        },
        "root": Object {
          "borderRadius": "4px",
        },
      }
    }
    target="#likeBtn12"
  >
    <Text>
       
      <Text>
        Thank you. Need to give
         
        <StyledLinkBase
          onClick={[Function]}
        >
          more feedback?
        </StyledLinkBase>
      </Text>
    </Text>
  </Callout>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikePressed",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn12"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn13"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should hover over pressed dislike and leave 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn22"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikePressed",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn23"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should hover over pressed like and leave 1`] = `
<Stack
  horizontal={true}
>
  <Callout
    directionalHint={1}
    onDismiss={[Function]}
    role="dialog"
    style={
      Object {
        "borderRadius": "4px",
        "padding": "5px 12px 5px 12px",
      }
    }
    styles={
      Object {
        "beakCurtain": Object {
          "borderRadius": "4px",
        },
        "root": Object {
          "borderRadius": "4px",
        },
      }
    }
    target="#likeBtn10"
  >
    <Text>
       
      <Text>
        Thank you. Need to give
         
        <StyledLinkBase
          onClick={[Function]}
        >
          more feedback?
        </StyledLinkBase>
      </Text>
    </Text>
  </Callout>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikePressed",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn10"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn11"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should hover over rest dislike and leave 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn20"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn21"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;

exports[`Feedback buttons snapshot tests should hover over rest like and leave 1`] = `
<Stack
  horizontal={true}
>
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
        },
      }
    }
    id="likeBtn8"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <CustomizedIconButton
    iconProps={
      Object {
        "imageProps": Object {
          "src": "LikeRest",
        },
        "style": Object {
          "minHeight": "18px",
          "transform": "rotate(180deg)",
        },
      }
    }
    id="dislikeBtn9"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</Stack>
`;
